packages/firebase_auth/firebase_auth_platform_interface/lib/src/types.dart

```diff
@@ -0,0 +1,242 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:meta/meta.dart';
+
+typedef void PhoneVerificationCompleted(AuthCredential phoneAuthCredential);
+typedef void PhoneVerificationFailed(AuthException error);
+typedef void PhoneCodeSent(String verificationId, [int forceResendingToken]);
+typedef void PhoneCodeAutoRetrievalTimeout(String verificationId);
+
+/// Represents a `User` in Firebase.
```

Comment:
Since these are all just data, and are mostly intended to be consumed by a "friend" package (`firebase_auth`) I think it's ok to leave them in this one file.

Reply:
OK